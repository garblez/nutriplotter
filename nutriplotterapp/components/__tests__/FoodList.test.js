import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import List from '../FoodList.js';
import popArray from '../../populateArray.js';
import popNameList from '../../populateNameList.js';
global.favourites = [];
//var r = new popNameList();
//var p = new popArray();

//first test to ensure working correctly
it('Test conditions working properly', async () => {
    const testVar = true;
    expect(true).toEqual(testVar);
});
//2 components in this filename, only List exported


//tests for list
it('List: renders correctly', () => {
  const tree = renderer.create(<List />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('List: State init ', () => {
  let ListComponent = renderer.create(<List />).getInstance();
  const targetState = {
    name: '',
    test: 'Recently Searched:',
    names: [],
    refresh: 0,
  };
  expect(ListComponent.state).toEqual(targetState);
});
