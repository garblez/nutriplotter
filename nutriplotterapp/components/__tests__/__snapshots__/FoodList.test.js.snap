// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List: renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "transparent",
        "borderBottomWidth": 1,
        "borderTopColor": "transparent",
        "borderTopWidth": 1,
        "padding": 8,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#bdc6cf",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "borderRadius": 3,
            "flexDirection": "row",
            "height": 30,
            "overflow": "hidden",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                  },
                }
              }
              underlayColor="white"
            >
              <Text />
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onClear={[Function]}
          onFocus={[Function]}
          placeholder="Enter a food"
          placeholderTextColor="#86939e"
          platform="default"
          style={
            Object {
              "alignSelf": "center",
              "color": "#86939e",
              "flex": 1,
              "fontSize": 18,
              "height": 40,
              "marginLeft": 10,
            }
          }
          testID="searchInput"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 15,
              "marginRight": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    >
      Recently Searched:
    </Text>
  </View>
</View>
`;
